server/sonar-ce/src/test/java/org/sonar/ce/cluster/HazelcastClientWrapperImplTest.java

```diff
@@ -0,0 +1,183 @@
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+package org.sonar.ce.cluster;
+
+import com.hazelcast.core.HazelcastInstance;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.Set;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.DisableOnDebug;
+import org.junit.rules.ExpectedException;
+import org.junit.rules.TestRule;
+import org.junit.rules.Timeout;
+import org.sonar.api.config.MapSettings;
+import org.sonar.api.config.PropertyDefinitions;
+import org.sonar.api.config.Settings;
+import org.sonar.process.NetworkUtils;
+import org.sonar.process.ProcessProperties;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.data.MapEntry.entry;
+
+public class HazelcastClientWrapperImplTest {
+  @Rule
+  public ExpectedException expectedException = ExpectedException.none();
+
+  @Rule
+  public TestRule safeGuard = new DisableOnDebug(Timeout.seconds(10));
+
+  private static HazelcastInstance hzCluster;
+  private static HazelcastClientWrapperImpl hzClient;
+
+  @BeforeClass
+  public static void setupHazelcastClusterAndHazelcastClient() {
+    int port = NetworkUtils.freePort();
+    hzCluster = HazelcastTestHelper.createHazelcastCluster("cluster_with_client", port);
+
+    Settings settings = createClusterSettings("cluster_with_client", "localhost:" + port);
+    hzClient = new HazelcastClientWrapperImpl(settings);
+  }
+
+  @AfterClass
+  public static void stopHazelcastClusterAndHazelcastClient() {
+    try {
+      hzClient.stop();
+    } catch (Exception e) {
+      // Ignore it
+    }
+    try {
+      hzCluster.shutdown();
+    } catch (Exception e) {
+      // Ignore it
+    }
+  }
+
+  @Test
+  public void constructor_throws_ISE_if_missing_CLUSTER_ENABLED_throw_ISE() {
+    Settings settings = createClusterSettings("sonarqube", "localhost:9003");
+    settings.removeProperty(ProcessProperties.CLUSTER_ENABLED);
+
+    expectedException.expect(IllegalStateException.class);
+    expectedException.expectMessage("Cluster is not enabled");
+
+    new HazelcastClientWrapperImpl(settings);
+  }
+
+  @Test
+  public void constructor_throws_ISE_if_missing_CLUSTER_NAME_throw_ISE() {
+    Settings settings = createClusterSettings("sonarqube", "localhost:9003");
+    settings.removeProperty(ProcessProperties.CLUSTER_NAME);
+
+    expectedException.expect(IllegalStateException.class);
+    expectedException.expectMessage("sonar.cluster.name is missing");
+
+    new HazelcastClientWrapperImpl(settings);
+  }
+
+  @Test
+  public void constructor_throws_ISE_if_missing_CLUSTER_LOCALENDPOINT_throw_ISE() {
+    Settings settings = createClusterSettings("sonarqube", "localhost:9003");
+    settings.removeProperty(ProcessProperties.CLUSTER_LOCALENDPOINT);
```

Comment:
the case where property `ProcessProperties.CLUSTER_LOCALENDPOINT` contains garbage is actually not tested

the case where property is empty is not tested either